cliquet/scripts/cliquet.py

```diff
@@ -15,6 +17,12 @@ def deprecated_init(env):
 
 def init_schema(env):
     registry = env['registry']
+    settings = registry.settings
+
+    if settings.get('read_only', False):
+        message = 'You cannot migrate the database with a read_only setup.'
+        warnings.warn(message, ReadOnlyError)
```

Comment:
Warnings should be used just to warn about something, and let us run python in a mode where a warning is turned into an exception.

Here, you are doing a warning and exiting no matter what. If this is the only use case for ReadOnlyError, you should turn it into an Exception and simply raise it. And remove the explicit sys.exit(1) code, because raising an error exits with 1.